ComponentKit/DataSources/CKCollectionViewTransactionalDataSource.mm

```diff
@@ -138,7 +158,15 @@ - (void)transactionalComponentDataSource:(CKTransactionalComponentDataSource *)d
     CKComponentBoundsAnimationApply(boundsAnimation, ^{
       for (NSIndexPath *indexPath in changes.updatedIndexPaths) {
         CKTransactionalComponentDataSourceItem *item = [state objectAtIndexPath:indexPath];
+          
+        // There is a race condition that causes this method to return nil
+        // between when the cell is first requested and when the cell is
+        // placed in the collection view. Thus we fall back on our mapping.
```

Comment:
This is something I noticed as a result of the optimization. If I issue two updates in quick succession, i.e. an insert and then an update, the update would not find a cell to mount, and silently fail.

Previously, updates were only performed after calling `-[UICollectionView performBatchUpdates:completion:]`, after which the issue doesn't seem to occur.